File: app/Http/Controllers/SpecialistPatternsJava/Lab1/Lab1Java/src/graph/Scene.java
package graph;

import java.util.ArrayList;
import java.util.List;

// Класс Scene представляет собой сцену для отображения графических объектов.
public class Scene {
    private List<GraphObject> objects; // Список объектов, отображаемых на сцене.
    public static final Scene instance = new Scene(); // Статический экземпляр сцены для доступа.

    // Приватный конструктор сцены, инициализирующий список объектов.
    private Scene() {
        objects = new ArrayList<GraphObject>();
    }

    // Метод для добавления объекта на сцену.
    public void add(GraphObject o) {
        objects.add(o);
    }

    // Метод для очистки сцены от всех объектов.
    public void clear() {
        objects.clear();
    }

    // Метод для отрисовки всех объектов на сцене.
    public void draw() {
        for (GraphObject g : objects) {
            g.draw();
        }
    }
}

File: app/Http/Controllers/SpecialistPatterns/LR/LR1SP/graph/Scene.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\LR\LR1SP\graph;


// Класс Scene представляет собой сцену, на которой размещаются графические объекты.
class Scene {
    private array $objects = [];
    public static Scene $instance;

    // Приватный конструктор для создания единственного экземпляра сцены.
    private function __construct() {
        $this->objects = [];
    }

    // Получение экземпляра сцены (реализация синглтона).
    public static function getInstance(): Scene {
        if (!isset(self::$instance)) {
            self::$instance = new Scene();
        }
        return self::$instance;
    }

    // Добавление графического объекта на сцену.
    public function add(GraphObject $o) {
        $this->objects[] = $o;
    }

    // Очистка сцены от объектов.
    public function clear() {
        $this->objects = [];
    }

    // Отрисовка всех объектов на сцене.
    public function draw() {
        foreach ($this->objects as $g) {
            $g->draw();
        }
    }
}
